contrib/rust/src/python/pants/contrib/rust/utils/basic_invocation_conversion_rules.py

```diff
@@ -0,0 +1,132 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import copy
+import os
+
+from pants.contrib.rust.targets.cargo_synthetic_custom_build import CargoSyntheticCustomBuild
+
+
+def args_rules(invocation_key, target, result_dir, crate_out_dirs, libraries_dir, make_dirs,
+               **kwargs):
+  def _c_flag_rules(key_value):
+    def _incremental(_):
+      new_path = os.path.join(result_dir, 'incremental')
+      make_dirs.update({new_path: True})
+      return new_path
+
+    rules = dict({
+      'incremental': _incremental,
+    })
+
+    is_key_value = key_value.split('=')
+
+    if len(is_key_value) == 2:
+      key, value = is_key_value
+      apply_rule = rules.get(key)
+      return (key + "=" + apply_rule(value)) if apply_rule else key_value
+    else:
+      return key_value
+
+  def _l_flag_rules(key_value):
+    def dependency(_):
+      return libraries_dir
+
+    rules = dict({
+      'dependency': dependency,
+    })
+
+    is_key_value = key_value.split('=')
+
+    if len(is_key_value) == 2:
+      key, value = is_key_value
+      apply_rule = rules.get(key)
+      return (key + "=" + apply_rule(value)) if apply_rule else key_value
+    else:
+      return key_value
+
+  def _out_dir_flag_rule(_):
+    new_path = os.path.join(result_dir, 'deps')
+    make_dirs.update({new_path: True})
+    return new_path
+
+  def _extern_flag_rule(key_value):
+    key, value = key_value.split('=')
+
+    for dependency in target.dependencies:
+      package_name, out_dir = crate_out_dirs[dependency.address.target_name]
+      if key == package_name:
+        return key + '=' + out_dir[0]
```

Comment:
We try to avoid concatenating strings. Instead, we use `.format()`.

This would be changed to:

```python
return "{}={}".format(key, out_dir[0])
```